src/objects/arrow_pickup.gd

```diff
@@ -0,0 +1,24 @@
+extends Area3D
+
+signal arrow_collected
+
+
+const SPIN_SPEED := 0.5
+const VERTICAL_OSCILLATION_SPEED := 1
+const VERTICAL_OSCILLATION_AMPLITUDE := 0.15
+
+@onready var mesh_instance_3d = $MeshInstance3D
+@onready var initial_y = mesh_instance_3d.position.y
+
+
+func _physics_process(delta) -> void:
+	mesh_instance_3d.rotation.y += SPIN_SPEED * delta
+	mesh_instance_3d.position.y = initial_y \
```

Comment:
Instead of doing this, just use a tween to animate the position. Sinusoidal interpolation is pretty easy to do.

Reply:
No